### DISCUSSION TITLE
Eloquent Relation Bug

### DISCUSSION THREAD

**Discussion by rcapulsuhay** [laravel/framework#51135] [Q&A] [answered]
### Laravel Version

11

### PHP Version

8.2

### Database Driver & Version

sqlsrv

### Description

the localID is 202416-0936 which is string but from some reason the eloquent removing the -0936 and turned into INT

This is IntLotNoModel
public function efai001s(): HasMany
    {
    //there are no foreignKey on EFAI001 , I just want to get rows from same column
        return $this->hasMany(EFAI001::class, 'intlot', 'intlot');
    }
    
    
SQL server error
SQLSTATE[22018]: [Microsoft][ODBC Driver 17 for SQL Server][SQL Server]Conversion failed when converting the varchar value '202238-2853' to data type int. (Connection: sqlsrv, SQL: select * from [efai001] where [efai001].[intlot] in (202416))

also, this works fine inside the controller

EFAI001::where('intlot', $intlot)->get();

### Steps To Reproduce

you must have 2 models but the unique key is like this 202416-0936 and it should be VARCHAR (50)

**Comment by ghost** [comment, marked as answer]
Did you define protected $keyType = 'string' in your model?

**Reply by rcapulsuhay** [reply]
thank you!!

my first try is adding it on relation model which still causing the error
now after trying to add it on the parent model the error is gone
so i need to define this keyType in every model if i have string primaryKey

i also thought that using protected $casts is enough

anyway thanks!
